cfpb/rental-assistance-finder
# Add i18n and Spanish translations

Add internationalization framework and Spanish translations.

## Additions

- i18next and react-i18next
- i18n config file
- Spanish & English translation files
- Get page language from app container `data-language` attribute

## Changes

- useTranslation hook instead of hard-coded English text

## Testing

1. Check out branch & run yarn start. To view with CFPB design system styles, copy contents of `/sample/cf.css` into `App.css`
2. Verify that app content displays as expected in English
3. Set `data-language` attribute to `es` in `index.html`
4. Verify that app content displays as expected in Spanish

## Screenshots
<img width="889" alt="Screen Shot 2021-08-12 at 1 43 24 PM" src="https://user-images.githubusercontent.com/778171/129267171-4e331b1e-b8ed-4ddb-9ec5-78087a3ee0ab.png">
<img width="856" alt="Screen Shot 2021-08-12 at 1 43 32 PM" src="https://user-images.githubusercontent.com/778171/129267182-89420a3e-e186-4075-83cb-e2e1150f912c.png">
<img width="840" alt="Screen Shot 2021-08-12 at 1 43 39 PM" src="https://user-images.githubusercontent.com/778171/129267190-09c3c772-0b31-460d-8d50-db2007d7c2a2.png">
<img width="837" alt="Screen Shot 2021-08-12 at 1 43 47 PM" src="https://user-images.githubusercontent.com/778171/129267196-8c964b12-268b-4c01-9027-5dbe010f7097.png">

## Todos

- As with most functionality, need to add tests
